mlx/transforms.cpp

```diff
@@ -655,6 +656,11 @@ std::vector<array> vmap_replace(
     }
 
     auto [v_outputs, v_out_axes] = a.primitive().vmap(v_inputs, v_axes);
+
+    assert(
+        "Expected the number of outputs and vmapped axes to match. Please fix "
+        "the vmap implementation for this primitive.");
```

Comment:
I'm not sure what is the purpose of this loc, maybe remove?

Reply:
I think this was `assert(v_outputs.size() == v_out_axes.size(), "...")` but I kind of agree that we should remove it.